rfcs/rfc-0001-lookup-by-username.md

```diff
@@ -0,0 +1,51 @@
+- Feature Name: (API route to look up user by username)
+- Start Date: (15/03/2023)
+- RFC PR: https://github.com/revoltchat/rfcs/issues/1
+- Tracking Issue: https://github.com/revoltchat/revolt/issues/0000
+- Status: draft
+
+# Summary
+
+Add an API route to look up users by username instead of ID
+
+# Motivation
+
+Currently, bots need to keep a cache of every user if they want to allow commands to target users by username (e.g. /ban @insert).
+I am proposing an API route that allows clients to look up users by username.
+
+# Guide-level explanation
+
+Send a `GET` request to `/users/by-name/insert` (no idea how to name the route) to get an array of users matching that username.
```

Comment:
This should probably be a POST to /users/by-name or something along those lines using the body to specify username and later discriminator.